src/core/system_parameters/beidou_cnav2_ephemeris.cc

```diff
@@ -0,0 +1,185 @@
+ * \author Dong Kyeong Lee, 2018. dole7890(at)colorado.edu
+ * \see <a href="http://m.beidou.gov.cn/xt/gfxz/201712/P020171226742357364174.pdf">beidou ICD</a>
+ *
+ * -------------------------------------------------------------------------
+ *
+ * Copyright (C) 2010-2015  (see AUTHORS file for a list of contributors)
+ *
+ * GNSS-SDR is a software defined Global Navigation
+ *          Satellite Systems receiver
+ *
+ * This file is part of GNSS-SDR.
+ *
+ * GNSS-SDR is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * GNSS-SDR is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GNSS-SDR. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * -------------------------------------------------------------------------
+ */
+
+#include "beidou_cnav2_ephemeris.h"
+#include "gnss_satellite.h"
+#include "BEIDOU_B2a_CA.h"
+#include <cmath>
+
+
+Beidou_Cnav2_Ephemeris::Beidou_Cnav2_Ephemeris()
+{
+
+    t_oe = 0.0;				//!< Ephemeris reference time [s]
+    SatType = 0.0;			//!< Satellite orbit type [dimensionless]
+    d_A = 0.0;				//!< Semi-major axis difference at reference time [m]
+    A_dot = 0.0;			//!< Change rate in semi-major axis [m/s]
+    dn_0 = 0.0;				//!< Mean motion difference from computed value at reference time [pi/s]
+    dn_0_dot = 0.0;			//!< Rate of mean motion difference from computed value at reference time [pi/s^2]
+    M_0 = 0.0;				//!< Mean anomaly at reference time [pi]
+    e = 0.0;				//!< Eccenticity [dimensionless]
+    omega = 0.0;			//!< Argument of perigee [pi]
+    Omega_0 = 0.0;			//!< Longitude of ascending node of orbital plane at weekly epoch [pi]
+    i_0 = 0.0;				//!< Inclination angle at reference time [pi]
+    Omega_dot = 0.0;		//!< Rate of right ascension [pi/s]
+    i_0_dot = 0.0;			//!< Rate of inclination angle [pi/s]
+    C_is = 0.0;				//!< Amplitude of sine harmonic correction term to the angle of inclination [rad]
+    C_ic = 0.0;				//!< Amplitude of cosine harmonic correction term to the angle of inclination [rad]
+    C_rs = 0.0;				//!< Amplitude of sine harmonic correction term to the orbit radius [m]
+    C_rc = 0.0;				//!< Amplitude of cosine harmonic correction term to the orbit radius [m]
+    C_us = 0.0;				//!< Amplitude of sine harmonic correction to the argument of latitude [rad]
+    C_uc = 0.0;				//!< Amplitude of cosine harmonic correction to the argument of latitude [rad]
+
+
+}
+
+
+boost::posix_time::ptime Beidou_Cnav2_Ephemeris::compute_beidou_time(const double offset_time) const
```

Comment:
Way too much copy/paste, if not using the time conversion functions should be removed. :)